spatialdata/_core/_spatialdata.py

```diff
@@ -0,0 +1,262 @@
+    labels: Mapping[str, Union[SpatialImage, MultiscaleSpatialImage]] = MappingProxyType({})
+    points: Mapping[str, AnnData] = MappingProxyType({})
+    polygons: Mapping[str, GeoDataFrame] = MappingProxyType({})
+    shapes: Mapping[str, AnnData] = MappingProxyType({})
+    _table: Optional[AnnData] = None
+
+    def __init__(
+        self,
+        images: Mapping[str, Any] = MappingProxyType({}),
+        labels: Mapping[str, Any] = MappingProxyType({}),
+        points: Mapping[str, Any] = MappingProxyType({}),
+        polygons: Mapping[str, Any] = MappingProxyType({}),
+        shapes: Mapping[str, Any] = MappingProxyType({}),
+        table: Optional[AnnData] = None,
+        **kwargs: Any,
+    ) -> None:
+
+        if images is not None:
+            self.images = {}
+            for k, v in images.items():
+                if ndim(v) == 3:
+                    Image2D_s.validate(v)
+                    self.images[k] = v
+                elif ndim(v) == 4:
+                    Image3D_s.validate(v)
+                    self.images[k] = v
+
+        if labels is not None:
+            self.labels = {}
+            for k, v in labels.items():
+                if ndim(v) == 2:
+                    Label2d_s.validate(v)
+                    self.labels[k] = v
+                elif ndim(v) == 3:
+                    Label3D_s.validate(v)
+                    self.labels[k] = v
+
+        if polygons is not None:
+            self.polygons = {}
+            for k, v in polygons.items():
+                Polygon_s.validate(v)
+                self.polygons[k] = v
+
+        if shapes is not None:
+            self.shapes = {}
+            for k, v in shapes.items():
+                Shape_s.validate(v)
+                self.shapes[k] = v
+
+        if points is not None:
+            self.points = {}
+            for k, v in points.items():
+                Point_s.validate(v)
+                self.points[k] = v
+
+        if table is not None:
+            Table_s.validate(table)
+            self._table = table
+
+    def write(self, file_path: str) -> None:
+        """Write to Zarr file."""
+
+        store = parse_url(file_path, mode="w").store
+        root = zarr.group(store=store)
+
+        # get union of unique ids of all elements
+        elems = set().union(
+            *[
+                set(i)
+                for i in [
+                    self.images,
+                    self.labels,
+                    self.points,
+                    self.polygons,
+                    self.shapes,
+                ]
+            ]
+        )
+
+        for el in elems:
+            elem_group = root.create_group(name=el)
+            if self.images is not None and el in self.images.keys():
+                write_image(
+                    image=self.images[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
```

Comment:
Can we use the `to_zarr()` methods on each element instead of the `write_*` functions? That way we just need to iterate over all of the elements. Something like:

```python
element_names = ["images", "labels", ...]
for element in element_names:
    element_dict = getattr(self, element)
    if len(element_dict) == 0:
         continue
    elem_group = root.create_group(name=element)
    for k, v in element_dict.items():
        v.to_zarr()
```



```

Reply:
Not sure we can, since the elements are of different classes and we need to have per-element control of what is being written (for example `chunks`, `compressor` etc for images and labels)